src/payment/models/buy-fiat/buy-fiat.service.ts

```diff
@@ -21,4 +31,125 @@ export class BuyFiatService {
 
     return await this.buyFiatRepo.save(entity);
   }
+
+  async update(id: number, dto: UpdateBuyFiatDto): Promise<BuyFiat> {
+    let entity = await this.buyFiatRepo.findOne(id, { relations: ['sell', 'sell.user'] });
+    if (!entity) throw new NotFoundException('Buy fiat not found');
+
+    const sellIdBefore = entity.sell?.id;
+
+    const update = this.buyFiatRepo.create(dto);
+
+    // buy
+    if (dto.sellId) update.sell = await this.getSell(dto.sellId);
+
+    // bank tx
+    if (dto.bankTxId) {
+      update.bankTx = await this.bankTxRepo.findOne({ id: dto.bankTxId });
+      if (!update.bankTx) throw new BadRequestException('Bank TX not found');
+    }
+
+    Util.removeNullFields(entity);
+
+    //TODO update aller Felder wieder deaktivieren
+    entity = await this.buyFiatRepo.save({ ...entity, ...update });
+
+    // activate user
+    if (entity.amlCheck === AmlCheck.PASS && entity.sell?.user?.status === UserStatus.NA) {
```

Comment:
Würde den Check `user?.status === UserStatus.NA` auch in das `activateUser()` verschieben.